File: totaleconomy-bukkit/src/main/java/com/ericgrandt/totaleconomy/commands/PayCommandExecutor.java
package com.ericgrandt.totaleconomy.commands;

import com.ericgrandt.totaleconomy.common.command.CommonParameter;
import com.ericgrandt.totaleconomy.common.command.PayCommand;
import com.ericgrandt.totaleconomy.commonimpl.BukkitPlayer;
import com.ericgrandt.totaleconomy.wrappers.BukkitWrapper;
import java.math.BigDecimal;
import java.util.Map;
import org.bukkit.command.Command;
import org.bukkit.command.CommandExecutor;
import org.bukkit.command.CommandSender;
import org.bukkit.entity.Player;
import org.jetbrains.annotations.NotNull;

public class PayCommandExecutor implements CommandExecutor {
    private final PayCommand payCommand;
    private final BukkitWrapper bukkitWrapper;

    public PayCommandExecutor(
        final PayCommand payCommand,
        final BukkitWrapper bukkitWrapper
    ) {
        this.payCommand = payCommand;
        this.bukkitWrapper = bukkitWrapper;
    }

    @Override
    public boolean onCommand(@NotNull CommandSender sender, @NotNull Command command, @NotNull String label, @NotNull String[] args) {
        if (!(sender instanceof Player player)) {
            return false;
        }

        Player toPlayer = bukkitWrapper.getPlayerExact(args[0]);
        String amountArg = args[1];
        if (!isValidDouble(amountArg)) {
            player.sendMessage("Invalid amount specified");
            return false;
        }

        Map<String, CommonParameter<?>> argsMap = Map.of(
            "toPlayer", new CommonParameter<>(new BukkitPlayer(toPlayer)),
            "amount", new CommonParameter<>(BigDecimal.valueOf(Double.parseDouble(amountArg)))
        );

        return payCommand.execute(new BukkitPlayer(player), argsMap);
    }

    private boolean isValidDouble(String amount) {
        try {
            Double.parseDouble(amount);
            return true;
        } catch (NumberFormatException e) {
            return false;
        }
    }
}

File: totaleconomy-paper/src/main/kotlin/com/ericgrandt/totaleconomy/commands/PayCommandExecutor.kt
package com.ericgrandt.totaleconomy.commands

import com.ericgrandt.totaleconomy.command.CommonParameter
import com.ericgrandt.totaleconomy.command.PayCommand
import com.ericgrandt.totaleconomy.impl.BukkitPlayer
import kotlinx.coroutines.CoroutineScope
import org.bukkit.Bukkit
import org.bukkit.command.Command
import org.bukkit.command.CommandExecutor
import org.bukkit.command.CommandSender
import org.bukkit.entity.Player

class PayCommandExecutor : CommandExecutor {
    private val scope: CoroutineScope
    private val payCommand: PayCommand

    constructor(scope: CoroutineScope, payCommand: PayCommand) {
        this.scope = scope
        this.payCommand = payCommand
    }

    override fun onCommand(
        sender: CommandSender,
        command: Command,
        label: String,
        args: Array<out String>,
    ): Boolean {
        var toPlayer = Bukkit.getPlayer(args[0])
        var amount = args[1].toDoubleOrNull()

        if (sender !is Player || toPlayer == null || amount == null) {
            return false
        }

        var argMap = mutableMapOf<String, CommonParameter>()
        argMap["toPlayer"] = CommonParameter.PlayerParam(BukkitPlayer(toPlayer))
        argMap["amount"] = CommonParameter.DoubleParam(amount)

        return payCommand.runAsync(scope, BukkitPlayer(sender), argMap)
    }
}
